sdk/trafficmanager/Azure.ResourceManager.TrafficManager/tests/Scenario/UserMetricsModelTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.TrafficManager.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.TrafficManager.Tests.Scenario
+{
+    public sealed class UserMetricsModelTests : TrafficManagerManagementTestBase
+    {
+        private SubscriptionResource _subscription;
+
+        public UserMetricsModelTests(bool isAsync) : base(isAsync) //, RecordedTestMode.Record)
+        { }
+
+        /// <summary>
+        /// The method creates a traffic manager profile, so it covers the creation use-case.
+        /// </summary>
+        /// <returns>A task to wait for.</returns>
+        [SetUp]
+        public async Task Setup()
+        {
+            _subscription = await Client.GetDefaultSubscriptionAsync();
+        }
+
+        [RecordedTest]
+        public async Task DeleteTest()
+        {
+            await Create();
+
+            UserMetricsModelResource userMetricsModelResource = _subscription.GetUserMetricsModel();
+            userMetricsModelResource = await userMetricsModelResource.GetAsync();
+
+            Assert.IsNotEmpty(userMetricsModelResource.Data.Key);
+
+            await Delete();
+
+            userMetricsModelResource = await userMetricsModelResource.GetAsync();
+
+            Assert.IsEmpty(userMetricsModelResource.Data.Key);
```

Comment:
validate that using a non existant resource results in `RequestFailedException` with a 404.

```
var badResource = Client.GetUserMetricsModelResource(new ResourceIdentifier(userMetricsModelResource.Id + "foo"));
var exception = Assert.ThrowsAsync<RequestFailedException>(async () => await badResource.DeleteAsync());
Assert.AreEqual(404, exception.Status);
```